drivers/system_pathways/app/models/system_pathways/chart_base.rb

```diff
@@ -74,31 +104,57 @@ def average(value, count)
     end
 
     private def filter_for_disabling_condition(scope)
-      return scope if @filter.disabling_condition.nil?
-
-      scope.merge(SystemPathways::Enrollment.where(disabling_condition: filter.disabling_condition))
+      scope = scope.merge(SystemPathways::Enrollment.where(disabling_condition: filter.disabling_condition)) unless filter.disabling_condition.nil?
+      scope = scope.merge(SystemPathways::Enrollment.where(disabling_condition: show_filter.disabling_condition)) unless show_filter&.disabling_condition.nil?
+      scope = scope.merge(SystemPathways::Enrollment.where(disabling_condition: details_filter.disabling_condition)) unless details_filter&.disabling_condition.nil?
+      scope
     end
 
     private def filter_for_chronic_at_entry(scope)
-      return scope if @filter.chronic_status.nil?
+      scope = scope.merge(SystemPathways::Enrollment.where(chronic_at_entry: filter.chronic_status)) unless filter.chronic_status.nil?
+      scope = scope.merge(SystemPathways::Enrollment.where(chronic_at_entry: show_filter.chronic_status)) unless show_filter&.chronic_status.nil?
+      scope = scope.merge(SystemPathways::Enrollment.where(chronic_at_entry: details_filter.chronic_status)) unless details_filter&.chronic_status.nil?
 
-      scope.merge(SystemPathways::Enrollment.where(chronic_at_entry: filter.chronic_status))
+      scope
     end
 
     private def filter_for_race(scope)
-      return scope unless filter.races.present?
-
       race_scope = nil
       filter.races.each do |column|
         next if column == 'MultiRacial'
 
         race_scope = add_alternative(race_scope, SystemPathways::Client.where(column.underscore.to_sym => true))
       end
-
       # Include anyone who has more than one race listed, anded with any previous alternatives
       race_scope ||= scope
       race_scope = race_scope.where(id: multi_racial_clients.select(:id)) if filter.races.include?('MultiRacial')
-      scope.merge(race_scope)
+      scope = scope.merge(race_scope)
+
+      race_scope = nil
```

Comment:
Maybe use something like `multiracial_scope`  above, and `show_filter_race_scope`/`detail_filte_race_scope` rather than re-using the same variable name?